pkg/analyze/kube_resource.go

```diff
@@ -125,15 +139,58 @@ func (a *AnalyzeClusterResource) analyzeResource(analyzer *troubleshootv1beta2.C
 		return nil, errors.Wrap(err, "failed to parse expected value as yaml doc")
 	}
 
-	result := &AnalyzeResult{
+	actualYAML, err := yaml.Marshal(actual)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal actual value")
+	}
+
+	if analyzer.ExpectedValue != "" {
```

Comment:
I think mentioning the order precedence of these parameters in the docs would come in handy.